File: lecture_006(2DArray)/l_008_diagonaltraversal.java
import java.util.Scanner;
public class l_008_diagonaltraversal {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        int n = scn.nextInt();
        int[][] arr = new int[n][n];
        input(arr);
        diagonalTraversal(arr);
    } 
    public static void display(int[][] arr) {
        for(int i=0; i<arr.length; i++) {
            for(int j=0; j<arr[0].length; j++) {
                System.out.print(arr[i][j]+" ");
            }
            System.out.println();
        }
    }
    public static void input(int[][] arr) {
        for(int i=0; i<arr.length; i++) {
            for(int j=0; j<arr[0].length; j++) {
                arr[i][j] = scn.nextInt();
            }
        }
    }
    public static void diagonalTraversal(int[][] arr) {
        for(int g=0; g<arr.length; g++) {
            for(int j=g, i=0; j<arr[0].length; j++,i++) {
                System.out.println(arr[i][j]);
            }
        }
    }
}

File: lecture_006(2DArray)/l_008_diagonaltraversal.cpp
#include<iostream>
#include<vector>
using namespace std;

void display(vector<vector<int>> &arr) {
    for(int i=0; i<arr.size(); i++) {
        for(int j=0; j<arr[0].size(); j++) {
            cout<<arr[i][j]<<" ";
        }
        cout<<endl;
    }
}
void input(vector<vector<int>> &arr) {
    for(int i=0; i<arr.size(); i++) {
        for(int j=0; j<arr[0].size(); j++) {
            cin>>arr[i][j];
        }
    }
}
void diagonaltraversal(vector<vector<int>> &arr) {
    for(int g=0; g<arr.size(); g++) {
        for(int j=g, i=0; j<arr[0].size(); j++,i++) {
            cout<<arr[i][j]<<endl;
        }
    }
}
int main() {
    int n;
    cin>>n;
    vector<vector<int>>arr(n,vector<int>(n,0));
    input(arr);
    diagonaltraversal(arr);
}